### PR TITLE
Convert bench/data/filters.json over to use filter expressions

### LINKED ISSUE #8079: Benchmark filter expressions
Looks like work got started for that in [this branch](https://github.com/mapbox/mapbox-gl-js/tree/filter-bench). Does it make sense though to replace the values in https://github.com/mapbox/mapbox-gl-js/blob/master/bench/data/filters.json with their expression equivalent filters?

### PR DESCRIPTION
**Pull request by tristen**
This PR converts the filter values used in benchmarks over to expressions.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] post benchmark scores

---
Closes #8079

### CODE REVIEW

**Comment by ryanhamley** [comment]
Such a big file change! It looks pretty good to me but might be helpful to get @mourner to take a look as well since he knows the expressions API better than I do

File: bench/data/filters.json [hunk 1]
```diff
@@ -1,1451 +1,2038 @@
 [
     {
-        "filter": [
+        "filter":[
```
Comment (mourner) [inline bench/data/filters.json]: Please let's keep the existing formatting conventions to minimize the diff.

**Comment by tristen** [comment]
<details>
<summary><strong>Here's those results</strong></summary>

screencapture-localhost-9966-bench-versions-2019-05-06-12_41_02-2
screencapture-localhost-9966-bench-versions-2019-05-06-12_41_02

</details>

**Comment by mourner** [comment]
@tristen can you take a look at what's up with the huge `FilterEvaluate` regression (last bench in the pic)? Is it something expected?

**Comment by mourner** [comment]
Also looks like `FunctionConvert` bench is no longer relevant because conversion is a noop for the new filters.

**Comment by tristen** [comment]
Ugh. Apologies for merging this prematurely. Following up in a pr.

**Pull request merged**